am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementServiceBenchmarks.java

```diff
@@ -0,0 +1,118 @@
+public class AccessManagementServiceBenchmarks {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @State(Scope.Thread)
+    @SuppressWarnings("LineLength")
+    public static class RecordState {
+        String resourceId = UUID.randomUUID().toString();
+        String accessorId = UUID.randomUUID().toString();
+
+        //load in mock FPL case data
+        JsonNode inputJson;
+
+        {
+            try {
+                inputJson = MAPPER.readTree(ClassLoader.getSystemResource("mock_case_data.json"));
+            } catch (IOException e) {
+                //NO-OP
+            }
+        }
+
+        //initialise service with datasource. Uses springboot DataSourceBuilder.
+        AccessManagementService service =
+            new AccessManagementService(DataSourceBuilder
+                .create()
+                .username("amuser@am-lib-test-aat")
+                .password("upF#99gf7RAZ?77H")
+                .url("jdbc:postgresql://am-lib-test-aat.postgres.database.azure.com:5432/am?user=amuser@am-lib-test-aat&password=upF#99gf7RAZ?77H&sslmode=require")
+                .build()
+            );
+
+        //need to define role constant and add to AAT (for role based)
+
+        //define resource constant
+        Resource resource = Resource.builder()
+            .resourceId(resourceId)
+            .type(ResourceDefinition.builder()
+                .serviceName("Service 1")
+                .resourceType("Resource Type 1")
+                .resourceName("resource")
+                .build())
+            .resourceJson(inputJson)
+            .build();
+
+        @Setup
+        public void doSetup() {
+            //create map of explicit access JsonPointer -> Permission
+            Map<JsonPointer, Set<Permission>> attributePermissions =
+                ImmutableMap.<JsonPointer, Set<Permission>>builder()
+                    .put(JsonPointer.valueOf(""), ImmutableSet.of(READ, CREATE))
+                    .put(JsonPointer.valueOf("/orders"), ImmutableSet.of(CREATE))
+                    .put(JsonPointer.valueOf("/orders/orderType"), ImmutableSet.of(READ))
+                    .put(JsonPointer.valueOf("/orders/emergencyProtectionOrderDirections"), ImmutableSet.of(DELETE))
+                    .put(JsonPointer.valueOf("/caseName"), ImmutableSet.of(UPDATE, CREATE, READ, DELETE))
+                    .put(JsonPointer.valueOf("/children"), ImmutableSet.of(READ))
+                    .put(JsonPointer.valueOf("/children/firstChild"), ImmutableSet.of(UPDATE))
+                    .put(JsonPointer.valueOf("/children/adoption"), ImmutableSet.of(READ, CREATE))
+                    .put(JsonPointer.valueOf("/applicant"), ImmutableSet.of(READ, CREATE))
+                    .put(JsonPointer.valueOf("/solicitor"), ImmutableSet.of(UPDATE, DELETE, CREATE))
+                    .build();
+
+            //create explicit access grant
+            service.grantExplicitResourceAccess(ExplicitAccessGrant.builder()
```

Comment:
My understanding is that explicit grant is performed before each method call performed by benchmark. Am I right?

We need to change it to do bulk upload before benchmark start / one per benchmark and clear DB in benchmark tear down.